### PR TITLE
add option to use peft for training

### PR DESCRIPTION
**Pull request by prathikr**
This PR adds an option to use peft for finetuning text-classification task. It is required to run meta-llama/Llama-2-7b-hf which we would like to add to our internal benchmarking pipelines. Also adds an explicit padding token which resolves padding issues.

### CODE REVIEW

**Comment by ArthurZucker** [comment]
Hey! I'm afraid we can't do this! As the doc for our exemple folder mentions:

> While we strive to present as many use cases as possible, the scripts in this folder are just examples. It is expected that they won’t work out-of-the box on your specific problem and that you will be required to change a few lines of code to adapt them to your needs. To help you with that, most of the examples fully expose the preprocessing of the data. This way, you can easily tweak them.

**Comment by askhade** [comment]
@ArthurZucker given we are adding an optional feature, can you explain what is the concern here? Thanks!

**Comment by LysandreJik** [comment]
Hey @prathikr, @askhade, thank you for your contribution, we appreciate you opening a PR!

Arthur references this specific README for the examples: https://github.com/huggingface/transformers/tree/main/examples#examples

And more specifically:

> Please discuss on the [forum](https://discuss.huggingface.co/) or in an [issue](https://github.com/huggingface/transformers/issues) a feature you would like to implement in an example before submitting a PR; we welcome bug fixes, **but since we want to keep the examples as simple as possible it's unlikely that we will merge a pull request adding more functionality at the cost of readability.**


We are aligned with you that adding examples leveraging PEFT and parameter efficient fine-tuning would be cool, but this kind of feature can very easily blow up the size of our examples which are arguably already way too long. 

While the feature you add is optional and therefore doesn't add "bloat" to the service itself, the examples' goal is really readability and adaptability; adding too many optional features unfortunately reduces this readability.

We appreciate your understanding!
